docs/specification/draft/client/elicitation.mdx

```diff
@@ -151,144 +256,237 @@ To request information from a user, servers send an `elicitation/create` request
+The same request could direct the user into an OAuth authorization flow, or a payment flow. The only difference is the URL and the message.
+
+**Request:**
 
 ```json
 {
   "jsonrpc": "2.0",
-  "id": 2,
+  "id": 3,
+  "method": "elicitation/create",
+  "params": {
+    "mode": "url",
+    "elicitationId": "550e8400-e29b-41d4-a716-446655440000",
+    "url": "https://mcp.example.com/ui/set_api_key",
+    "message": "Please provide your API key to continue."
+  }
+}
+```
+
+**Response:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 3,
   "result": {
-    "action": "decline"
+    "action": "accept"
   }
 }
 ```
 
-**Cancel Response Example:**
+The response with `action: "accept"` indicates that the user has consented to the
+interaction. It does not mean that the interaction is complete. The interaction occurs out
+of band and the client is not aware of the outcome, unless the client requests progress updates.
+
+### Progress Tracking
+
+The client **MAY** request progress updates from the server by sending an `elicitation/track` request with a [progress token](../basic/utilities/progress). This is particularly useful in URL mode, because the client is not involved in the interaction.
+
+The client **MUST** include an `elicitationId` in the `elicitation/track` request to identify which elicitation to send progress updates for. The client **SHOULD** include a `progressToken` in the request's `_meta` field.
+
+The server **MUST** respond with an [error](#error-handling) if the client sends an `elicitation/track` request containing an `elicitationId` that is not known or does not belong to the client.
+
+After the client has requested progress updates via an `elicitation/track` request, the server **MAY** send [progress notifications](../basic/utilities/progress) to the client.
+
+The server **SHOULD** send a `elicitation/track` response to the client once the elicitation is complete.
+
+#### Example
+
+**Progress Tracking Request:**
 
 ```json
 {
   "jsonrpc": "2.0",
-  "id": 2,
+  "id": 4,
+  "method": "elicitation/track",
+  "params": {
+    "_meta": {
+      "progressToken": "abc123"
+    },
+    "elicitationId": "550e8400-e29b-41d4-a716-446655440000"
+  }
+}
+```
+
+**Progress Tracking Notification:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "method": "notifications/progress",
+  "params": {
+    "progressToken": "abc123",
+    "progress": 42,
+    "message": "Consent pending..."
+  }
+}
+```
+
+**Progress Tracking Response:**
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 4,
   "result": {
-    "action": "cancel"
+    "status": "complete"
+  }
+}
+```
+
+### Elicitation Required Error
+
+When another request cannot be processed until an elicitation is completed, the server **SHOULD** return an [ElicitationRequired error](#error-handling) (code `-32000`) to indicate to the client that an elicitation is required.
```

Comment:
```suggestion
When a request cannot be processed or suspended until an elicitation is completed, the server **MAY** return an [ElicitationRequired error](#error-handling) (code `-32000`) to indicate to the client that an elicitation is required.
```

What does "another" mean in "another request"? Shouldn't it just be "a request"?

And does the `**SHOULD**` here imply this is preferred over the `elicitation/create` request? I think `MAY` is better, because if a server can send a request rather than an error response, it could keep the operation ongoing and return a successful response immediately when the elicitation completes without the client having to optimistically retry or do any tracking.

Reply:
Agreed, these are good updates.